File: client/src/app/entities/sale.ts
import {User} from "./user";
import {DataPage} from "../shared/data-page";
import {Customer} from "./customer";
import {Saleitem} from "./saleitem";

export class Sale {
  id: number;
  code: string;
  tocreation: string;
  description: string;
  creator: User;
  customer: Customer;
  date: string;
  total: number;

  saleitemList: Saleitem[];

  constructor(id: number=null) {
    this.id = id;
  }
}
export class SaleDataPage extends DataPage{
  content: Sale[];
}

File: server/src/main/java/bit/project/server/entity/Sale.java
package bit.project.server.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
public class Sale {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDateTime tocreation;

    private String description;

   @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;

   @ManyToOne
    private Customer customer;

    private Date date;

    private BigDecimal total;


    @OneToMany(mappedBy = "sale",cascade = CascadeType.ALL,orphanRemoval = true)
    private List<Saleitem> saleitemList;

    public Sale(Integer id) {
        this.id = id;
    }
}
